### PR TITLE
add integration with leadsquared

### LINKED ISSUE #649: feat: Add integration with: LeadSquared (CRM)
### What type of software do you want to build integrations with?

CRM

### Which specific platforms would you integrate with?

[LeadSquared](https://www.leadsquared.com/us/)

### Which specific objects would you integrate with?

- [ ] Contacts
- [ ] Deals
- [ ] Tasks
- [ ] Engagements

### Do you anticipate to mostly read, or write data?

Read and write

### Please share any detail you want here

Documentation: https://apidocs.leadsquared.com/overview/#api

### PR DESCRIPTION
**Pull request by Iamsidar07**
fixes #649 

Integrated Objects:
Task,
Contacts,
Engagements,
Deal

## Summary by CodeRabbit

- **New Features**
  - Introduced a comprehensive integration with LeadSquared CRM, allowing users to manage contacts, deals, engagements, and tasks effectively.
  - Added synchronization capabilities for contacts, deals, engagements, and tasks to enhance data consistency.

- **Type Definitions**
  - Implemented new TypeScript types for improved type safety across LeadSquared-related data interactions.

- **Configuration Updates**
  - Updated metadata to include a new connector for LeadSquared, enhancing integration capabilities.

### CODE REVIEW

**Comment by rflihxyz** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview 783d0240-ec38-4387-a9a9-8e225f1bd3e1 | [dev](https://console.qovery.com/organization/e9b5deb5-34d8-4f9c-add2-4041675753df/project/30557e05-54b1-497e-a0a5-f9b1592a5b24/environment/783d0240-ec38-4387-a9a9-8e225f1bd3e1/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Truth can only be found in one place: the code

File: packages/api/src/crm/contact/services/leadsquared/types.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
+type LeadSquaredContact = {
+  /*
+   * user_id
+   * first_name
+   * last_name
+   * email
+   * phone
+   * address
+   */
```
[the remark this answers is not in the document]
Comment (Iamsidar07) [reply packages/api/src/crm/contact/services/leadsquared/types.ts:9]: ok

File: packages/api/src/crm/engagement/services/leadsquared/types.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+type SenderType =
+  | 'UserId' // Sender is a user ID
+  | 'UserEmailAddress' // Sender is a user email address
+  | 'APICaller' // Sender = ""
+  | 'LeadOwner'; // Sender = ""
+
+type RecipientType =
+  | 'LeadEmailAddress' // Recipient is a email address
+  | 'Lead Number' // Recipient is lead number
+  | 'LeadId'; // Recipient is lead id
```
[the remark this answers is not in the document]
Comment (Iamsidar07) [reply packages/api/src/crm/engagement/services/leadsquared/types.ts:37]: It should be like as it it is in docs

File: packages/shared/src/connectors/metadata.ts [hunk 3]
```diff
@@ -368,6 +368,30 @@ export const CONNECTORS_METADATA: ProvidersConfig = {
...
+              valueName: 'access_token',
+            },
+            {
+              headerParamName: 'x-LSQ-SecretKey',
+              valueName: 'secret_token',
+            },
+          ],
+          properties: ['access_token', 'secret_token']
+        },
+        active: false
```
[the remark this answers is not in the document]
Comment (Iamsidar07) [reply packages/shared/src/connectors/metadata.ts:394]: I'm not completly sure, It has to be work done on it.

File: packages/api/src/crm/engagement/services/leadsquared/mappers.ts [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+          (mapping) => mapping.slug === k,
+        );
+        if (mapping) {
+          result[mapping.remote_id] = v;
+        }
+      }
+    }
+
+    return result;
+  }
```
[the remark this answers is not in the document]
Comment (Iamsidar07) [reply packages/api/src/crm/engagement/services/leadsquared/mappers.ts:239]: It does not have source_number property

**Pull request closed without merging**